server/src/domain/database/database.service.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { newDatabaseRepositoryMock } from '@test';
+import { Version } from '../domain.constant';
+import { DatabaseExtension, IDatabaseRepository } from '../repositories';
+import { DatabaseService } from './database.service';
+
+describe(DatabaseService.name, () => {
+  let sut: DatabaseService;
+  let databaseMock: jest.Mocked<IDatabaseRepository>;
+
+  beforeEach(async () => {
+    databaseMock = newDatabaseRepositoryMock();
+
+    sut = new DatabaseService(databaseMock);
+    sut.minVectorsVersion = new Version(0, 1, 1);
+    sut.maxVectorsVersion = new Version(0, 1, 11);
+  });
+
+  it('should work', () => {
+    expect(sut).toBeDefined();
+  });
+
+  describe('init', () => {
+    beforeEach(() => {
+      sut.assertVectors = jest.fn();
```

Comment:
You should never mock sut methods, only repository methods.

Reply:
I just did that to avoid duplicating the tests. Should I copy/paste the assertion tests here too or is there another way to avoid duplicating them?

Reply:
Maybe a spy that just checks that it runs?

Reply:
Not sure what you mean by duplicating the tests.

Reply:
My thought for `init` was to specifically test that it calls those methods without being concerned about their behavior (because that's covered in other tests). So if something isn't mocked, then that means it's part of what's being tested and would need tests as a result. But I see where you're coming from with the domain/infra split so I can make that change.